components/tidb_query_executors/src/fast_hash_aggr_executor.rs

```diff
@@ -338,6 +369,8 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for FastHashAggregationImp
             }
         };
 
+        self.alloc_trace(self.states_offset_each_logical_row.len() * size_of::<usize>());
```

Comment:
Is it counted multiple times? Every time we call `process_batch_input`, we add its size to the total consumed memory. But it's cleared at the first of each `process_batch_input` call.

Reply:
It subtracts the previous size now.